### DISCUSSION TITLE
Examples for more complex applications (monorepo, next.js)

### DISCUSSION THREAD

**Discussion by erosenberg** [0no-co/gql.tada#367] [Ideas]
Is it possible to add more entries to the examples folder? These come to mind:

- managing multiple schemas in a monorepo
- Next.js application (using CLI to generate on local server start)
- an example with more env variables at play
- managing multiple environments' schemas in a single project (and how that would look when exporting`graphql` manually.
- examples using react-query

**Comment by kitten** [comment]
I'm a little conflicted about this, but generally not opposed. The problem is that this could get a little confusing:
- currently, `tsconfig.json#plugins` seems to specifially cause problems in our setup. Until that's figured out this will be a little weird
- `examples/` being part of the main monorepo has its benefits but also drawbacks
  - in `urql` we actually keep a workspace around for all examples in the subfolder. This has some benefits. But scripts are needed to keep some stuff in sync, which isn't set up in this repo yet
- `examples/` itself containing monorepos could become confusing
- the lack of an online sandbox (like Codesandbox and Stackblitz) supporting TypeScript plugins makes it hard to quickly try out any examples, and might make this entire effort a little less valuable

**Reply by erosenberg** [reply]
Oh, yeah I was pitching to create an example monorepo in the `examples/` folder, but I can see how that would be confused. Thanks for explaining!

**Comment by jmn90** [comment]
What about the last point specifically? Would like to see a recommended way to use gql.tada with react-query. Does not have to be in `/examples` for now, but maybe some quick recommendations in the documentation?
